File: week_10/c_sharp_course/week_1/day_4/composition/starter_code/wizard_management_specs/WizardTest.cs
using NUnit.Framework;

namespace WizardManagement { 

  [TestFixture]
  public class WizardTest {
    Wizard wizard;
    Broomstick broomstick;

    [SetUp]
    public void Init() 
    {
      broomstick = new Broomstick("Nimbus", 10);
      wizard = new Wizard("Toby", broomstick);
    }

    [Test]
    public void HasName()
    {
      Assert.AreEqual("Toby", wizard.Name);
    }

    [Test]
    public void HasBroomstick()
    {
      Assert.AreEqual("Nimbus", wizard.Broomstick.Brand);
    }

    [Test]
    public void CanFly()
    {
      Assert.AreEqual("mounting broom, running, skipping, flying!", wizard.Fly());
    }
  }
}

File: week_06/day_3/composition/Composition_Start/app/src/test/java/example/codeclan/com/composition_start/WizardTest.java
package example.codeclan.com.composition_start;

/**
 * Created by user on 28/08/2017.
 */

import static org.junit.Assert.*;
import org.junit.*;

import example.codeclan.com.composition_start.wizard_management.*;

public class WizardTest {

    Wizard wizard;
    Broomstick broomstick;

    @Before
    public void before(){
        broomstick = new Broomstick("Nimbus", 10);
        wizard = new Wizard("Toby", broomstick);
    }

    @Test
    public void hasName(){
        assertEquals("Toby", wizard.getName());
    }

    @Test
    public void hasBroomstick(){
        assertEquals("Nimbus", wizard.getBroomstick().getBrand());
    }

    @Test
    public void canFly(){
        assertEquals("mounting broom, running, skipping, flying!", wizard.fly());
    }
}
